cypress/support/validations/undefinedValidation.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @module undefinedValidation
+ * @function undefinedValidation
+ * @description To validate the content of the response if undefined
+ * @param {Object} validationTypeObject - The object consists of required fields for validation.
+ * @param {Object} apiOrEventObject - Extracted API or Event object from global list.
+ * @example
+ * cy.undefinedValidation(validationTypeObject, apiOrEventObject)
+ */
+
+Cypress.Commands.add('undefinedValidation', (validationTypeObject, apiOrEventObject) => {
+  if (apiOrEventObject != null && validationTypeObject != null) {
+    try {
+      validationTypeObject.validations.forEach((validation) => {
+        const fieldPath = validation.field.replace('result.', '');
+        const fieldParts = fieldPath.split('.');
+        const fieldValue = fieldParts.reduce((obj, part) => {
+          return obj && obj[part];
+        }, apiOrEventObject);
+        assert.isUndefined(fieldValue, validation.description);
+      });
+    } catch (error) {
+      cy.log(
```

Comment:
No need to check and loop through each field part after splitting with . 
The field part should be 
**result.styles.windowColor**


**assert.isUndefined(apiResponse.result.styles.windowColor, validation.description);**

Reply:
But if above looping is done just for extra precaution, then I am okay with the changes

Reply:
I dont think the above assertion works because value of field, eg: 'result.styles.windowColor' is a string and we cant assert apiOrEventObject.**'result.styles.windowColor'** right? That's the reason why its converted to array, looped through individual field properties and processed
```
  "ACCESSIBILITY_WINDOWCOLOR_VALUE_FOR_NULL_PARAMS": {
        "method": "accessibility.closedCaptionsSettings",
        "data": [
            {
                "type": "undefined",
                "validations": [
                    {
                        "field": "result.styles.windowColor",
                        "description": "Validation of the accessibility window color undefined"
                    }
                ]
            }
        ]
    },
```
What the above logic does is if field is 'result', directly assert if result is undefined, else it splits the nested field properties, traverses the nested structure of apiorEventObject to access the field value using reduce function..eg; apioreventobject[styles][windowcolor] and check if that value is undefined

Reply:
Also curious on this - how are we handling it for `validationJsonPath` like here: https://github.com/rdkcentral/firebolt-certification-suite/blob/dev/cypress/fixtures/fireboltCalls/accessibility.json#L476

Reply:
Have optimised the logic bit more and added comments for better understanding. 
So in the current implementation, we simply use eval js function to handle/extract validationJsonPath, eg: https://github.com/rdkcentral/firebolt-certification-suite/blob/db0214f1a1f27dc7d2bc3d87ebe8eefc17b0544a/cypress/support/cypress-commands/assertion.js#L148
Similarly, we can simply use something like that here as well which will be more simpler codewise
```
 let expected = eval('apiOrEventObject.response' + validation.field);
  assert.isUndefined(expected, validation.description);
```
but since eval is not a secure practice and there  is no proper error handling and logging in current case, I have implemented this in a more optimised way. Let me know which approach to move ahead with. Currently I havent been able to test it in xumo due to unavailabilty of device, but have done a first level testing of base logic in browser and that worked fine
cc @neeradanelxsi @jbigel